src/utilities/with-app-provider.tsx

```diff
@@ -52,66 +52,32 @@ export function withAppProvider<OwnProps>({withinScrollable}: Options = {}) {
-            {(link) => (
-              <ThemeProviderContext.Consumer>
-                {(theme) => (
-                  <I18nContext.Consumer>
-                    {(intl) => (
-                      <ScrollLockManagerContext.Consumer>
-                        {(scrollLockManager) => (
-                          <StickyManagerContext.Consumer>
-                            {(stickyManager) => (
-                              <AppBridgeContext.Consumer>
-                                {(appBridge) => {
-                                  const polarisContext: PolarisContext = {
-                                    link,
-                                    intl,
-                                    scrollLockManager,
-                                    stickyManager,
-                                    theme,
-                                    appBridge,
-                                  };
+      const polarisContext: PolarisContext = {
+        link,
+        intl,
+        scrollLockManager,
+        stickyManager,
+        theme,
+        appBridge,
+      };
 
-                                  if (
-                                    !intl ||
-                                    !scrollLockManager ||
-                                    !stickyManager ||
-                                    !link ||
-                                    !theme
-                                  ) {
-                                    throw new Error(
-                                      `The <AppProvider> component is required as of v2.0 of Polaris React. See https://polaris.shopify.com/components/structure/app-provider for implementation instructions.`,
-                                    );
-                                  }
-
-                                  return (
-                                    <WrappedComponent
-                                      {...this.props as any}
-                                      polaris={polarisContext}
-                                    />
-                                  );
-                                }}
-                              </AppBridgeContext.Consumer>
-                            )}
-                          </StickyManagerContext.Consumer>
-                        )}
-                      </ScrollLockManagerContext.Consumer>
-                    )}
-                  </I18nContext.Consumer>
-                )}
-              </ThemeProviderContext.Consumer>
-            )}
-          </LinkContext.Consumer>
+      if (!intl || !scrollLockManager || !stickyManager || !link || !theme) {
+        throw new Error(
+          `The <AppProvider> component is required as of v2.0 of Polaris React. See https://polaris.shopify.com/components/structure/app-provider for implementation instructions.`,
         );
       }
-    }
+
+      return <WrappedComponent {...props as any} polaris={polarisContext} />;
+    };
+    WithProvider.contextTypes = WrappedComponent.contextTypes;
```

Comment:
This is just for [legacy context](https://reactjs.org/docs/legacy-context.html) which we no longer us internally but [external users might](https://github.com/Shopify/web/search?q=withAppProvider&unscoped_q=withAppProvider)